### PR TITLE
[MIG][17.0] kpi

### PR DESCRIPTION
**Pull request by EdgarRetes**
@max3903

### CODE REVIEW

**Comment by EdgarRetes** [comment]
@max3903

**Comment by max3903** [comment]
@EdgarRetes I get this error message when creating a threshold:
```
RPC_ERROR
Odoo Server Error

Traceback (most recent call last):
  File "/opt/odoo/odoo/http.py", line 1782, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/opt/odoo/odoo/service/model.py", line 133, in retrying
    result = func()
  File "/opt/odoo/odoo/http.py", line 1809, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/opt/odoo/odoo/http.py", line 2013, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 221, in _dispatch
    result = endpoint(**request.params)
  File "/opt/odoo/odoo/http.py", line 757, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/opt/odoo/addons/web/controllers/dataset.py", line 24, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/opt/odoo/addons/web/controllers/dataset.py", line 20, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 468, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 453, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/opt/odoo/addons/web/models/models.py", line 73, in web_save
    self = self.create(vals)
  File "<decorator-gen-162>", line 2, in create
  File "/opt/odoo/odoo/api.py", line 391, in _model_create_single
    return create(self, arg)
  File "/mnt/data/odoo-addons-dir/kpi/models/kpi_threshold.py", line 65, in create
    for range1 in data["range_ids"][0][2]:
IndexError: list index out of range
```

File: kpi/static/src/img/icon.png
Comment (max3903) [inline kpi/static/src/img/icon.png:1]: @EdgarRetes Can you update the icon?

**Comment by EdgarRetes** [comment]
> @EdgarRetes I get this error message when creating a threshold:
> 
> ```
> RPC_ERROR
> Odoo Server Error
> 
> Traceback (most recent call last):
>   File "/opt/odoo/odoo/http.py", line 1782, in _serve_db
>     return service_model.retrying(self._serve_ir_http, self.env)
>   File "/opt/odoo/odoo/service/model.py", line 133, in retrying
>     result = func()
>   File "/opt/odoo/odoo/http.py", line 1809, in _serve_ir_http
>     response = self.dispatcher.dispatch(rule.endpoint, args)
>   File "/opt/odoo/odoo/http.py", line 2013, in dispatch
>     result = self.request.registry['ir.http']._dispatch(endpoint)
>   File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 221, in _dispatch
>     result = endpoint(**request.params)
>   File "/opt/odoo/odoo/http.py", line 757, in route_wrapper
>     result = endpoint(self, *args, **params_ok)
>   File "/opt/odoo/addons/web/controllers/dataset.py", line 24, in call_kw
>     return self._call_kw(model, method, args, kwargs)
>   File "/opt/odoo/addons/web/controllers/dataset.py", line 20, in _call_kw
>     return call_kw(request.env[model], method, args, kwargs)
>   File "/opt/odoo/odoo/api.py", line 468, in call_kw
>     result = _call_kw_multi(method, model, args, kwargs)
>   File "/opt/odoo/odoo/api.py", line 453, in _call_kw_multi
>     result = method(recs, *args, **kwargs)
>   File "/opt/odoo/addons/web/models/models.py", line 73, in web_save
>     self = self.create(vals)
>   File "<decorator-gen-162>", line 2, in create
>   File "/opt/odoo/odoo/api.py", line 391, in _model_create_single
>     return create(self, arg)
>   File "/mnt/data/odoo-addons-dir/kpi/models/kpi_threshold.py", line 65, in create
>     for range1 in data["range_ids"][0][2]:
> IndexError: list index out of range
> ```

I wasn't able to recreate the error, I didn't have any problems creating a threshold. Can you check again if it works with the last changes?

File: kpi/static/src/img/icon.png
Comment (EdgarRetes) [reply kpi/static/src/img/icon.png:1]: Where can I find the updated icon?

Comment (max3903) [reply kpi/static/src/img/icon.png:1]: https://github.com/OCA/management-system/tree/17.0/mgmtsystem/static/description

**Comment by max3903** [comment]
@EdgarRetes It still doesn't work. You can try here:
http://oca-reporting-engine-17-0-pr936-e4015e27de8e.runboat.odoo-community.org/web#action=183&model=kpi.threshold&view_type=list&cids=1&menu_id=124

File: kpi/views/kpi_views.xml [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        <field name="search_view_id" ref="view_kpi_filter" />
+    </record>
+    <record id="action_view_kpi_kanban" model="ir.actions.act_window.view">
+        <field eval="1" name="sequence" />
+        <field name="view_mode">kanban</field>
+        <field name="view_id" ref="view_kpi_kanban" />
+        <field name="act_window_id" ref="open_kpi_dashboard" />
+    </record>
+    <record model="ir.actions.act_window" id="open_kpi_list">
+        <field name="name">KPI Maintenance</field>
```
Comment (max3903) [inline kpi/views/kpi_views.xml]:
```suggestion
        <field name="name">KPI</field>
```

**Review by max3903 — APPROVED** [review]

File: kpi/views/kpi_views.xml (hunk 1 above)
Comment (EdgarRetes) [reply kpi/views/kpi_views.xml]: Done

**Pull request merged**